packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -912,9 +912,10 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       oldWidget.controller.removeListener(_didChangeTextEditingValue);
       widget.controller.addListener(_didChangeTextEditingValue);
       _updateRemoteEditingValueIfNeeded();
-    }
-    if (widget.controller.selection != oldWidget.controller.selection) {
+    } else if (widget.controller.selection != oldWidget.controller.selection) {
```

Comment:
Why else if? Don't we need to call _selectionIOverlay.update if the controller and the selection on it changed? (we should add a test to verify that this works)